lib/common/common/Int64Math.cpp

```diff
@@ -30,6 +37,16 @@ Int64Math::Mul(int64 left, int64 right, int64 *pResult)
     return (left != 0 && right != 0 && (*pResult / left) != right);
 #endif
 }
+#else
+// FIXME: this probably needs checks for __int128 at least.
+bool
+Int64Math::Mul(int64 left, int64 right, int64 *pResult)
+{
+    __int128 result = (__int128)left * (__int128)right;
+    *pResult = (int64)result;
+    return (left != 0 && right != 0 && (*pResult / left) != right);
```

Comment:
@xanlpz @dilijev In case you didn't notice, @curtisman proposed a much simpler solution above. All you need to do is changing line 31 from
```
#if _M_X64
```
to sth. like
```
#if defined(_MSC_VER) && defined(_M_X64)
```
Let the existing ```#else``` block handle all default cases.